src/S3SparseIterator.cpp

```diff
@@ -12,33 +12,32 @@
 namespace cirrus {
   
 // s3_cad_size nmber of samples times features per sample
-S3SparseIterator::S3SparseIterator(
-        uint64_t left_id, uint64_t right_id, // right id is exclusive
-        const Configuration& c,
-        uint64_t s3_rows,
-        uint64_t minibatch_rows,
-        bool use_label,
-        int worker_id,
-        bool random_access) :
-    left_id(left_id), right_id(right_id),
-    conf(c), s3_rows(s3_rows),
-    minibatch_rows(minibatch_rows),
-    //pm(REDIS_IP, REDIS_PORT),
-    minibatches_list(100000),
-    use_label(use_label),
-    worker_id(worker_id),
-    re(worker_id),
-    random_access(random_access)
-{
-      
+S3SparseIterator::S3SparseIterator(uint64_t left_id,
+                                   uint64_t right_id,  // right id is exclusive
+                                   const Configuration& c,
+                                   uint64_t s3_rows,
+                                   uint64_t minibatch_rows,
+                                   bool use_label,
+                                   int worker_id,
+                                   bool random_access)
+    : S3Iterator(c),
+      left_id(left_id),
+      right_id(right_id),
+      s3_rows(s3_rows),
+      minibatch_rows(minibatch_rows),
+      // pm(REDIS_IP, REDIS_PORT),
+      minibatches_list(100000),
+      use_label(use_label),
+      worker_id(worker_id),
+      re(worker_id),
+      random_access(random_access) {
   std::cout << "S3SparseIterator::Creating S3SparseIterator"
     << " left_id: " << left_id
     << " right_id: " << right_id
     << " use_label: " << use_label
     << std::endl;
 
   // initialize s3
-  s3_initialize_aws();
```

Comment:
Instead of remove this, I would say keep this here incase s3 is not active already. By fixing the s3_initialize_aws function as I suggested above, this would never throw an std::runtime_error.